pkgs/leak_tracker_flutter_testing/lib/src/leak_tracking_for_tests.dart

```diff
@@ -0,0 +1,222 @@
+        collectRetainingPathForNotGCed: true,
+      ),
+    );
+  }
+
+  /// Adds certain classes and leak types to ignore lists in [settings].
+  static void ignore({
+    Map<String, int?> notGCed = const {},
+    bool allNotGCed = false,
+    Map<String, int?> notDisposed = const {},
+    bool allNotDisposed = false,
+    List<String> classes = const [],
+  }) {
+    settings = withIgnoredLeaks(
+      notDisposed: notDisposed,
+      notGCed: notGCed,
+      allNotDisposed: allNotDisposed,
+      allNotGCed: allNotGCed,
+      classes: classes,
+    );
+  }
+
+  /// Returns copy of [settings] with extended skip lists.
+  ///
+  /// In the result the skip limit for a class is maximum of two original skip limits.
+  /// Items in [classes] will be added to all skip lists.
+  static LeakTrackingForTestsSettings withIgnoredLeaks({
+    Map<String, int?> notGCed = const {},
+    bool allNotGCed = false,
+    Map<String, int?> notDisposed = const {},
+    bool allNotDisposed = false,
+    List<String> classes = const [],
+  }) {
+    Map<String, int?> addClassesToMap(
+      Map<String, int?> map,
+      List<String> classes,
+    ) {
+      return {
+        ...map,
+        for (final c in classes) c: null,
+      };
+    }
+
+    return settings.copyWith(
+      ignoredLeaks: IgnoredLeaks(
+        notGCed: settings.ignoredLeaks.notGCed.merge(
+          IgnoredLeaksSet(
+            byClass: addClassesToMap(notGCed, classes),
+            ignoreAll: allNotGCed,
+          ),
+        ),
+        notDisposed: settings.ignoredLeaks.notDisposed.merge(
+          IgnoredLeaksSet(
+            byClass: addClassesToMap(notDisposed, classes),
+            ignoreAll: allNotDisposed,
+          ),
+        ),
+      ),
+    );
+  }
+
+  /// Returns copy of [settings] with reduced ignore lists.
+  ///
+  /// Items in [classes] will be removed from all ignore lists.
+  static LeakTrackingForTestsSettings copyWithTrackedLeaks({
+    List<String> notGCed = const [],
+    List<String> notDisposed = const [],
+    List<String> classes = const [],
+  }) {
+    final result = settings.copyWith(
+      ignoredLeaks: IgnoredLeaks(
+        notGCed: settings.ignoredLeaks.notGCed.track([...notGCed, ...classes]),
+        notDisposed: settings.ignoredLeaks.notDisposed
+            .track([...notDisposed, ...classes]),
+      ),
+    );
+    return result;
+  }
+
+  /// Removes certain classes and leak types from ignore lists in [settings].
+  ///
+  /// Items in [classes] will be removed from all ignore lists.
+  static void trackLeaks({
+    List<String> notGCed = const [],
+    List<String> notDisposed = const [],
+    List<String> classes = const [],
+  }) {
+    settings = copyWithTrackedLeaks(
+      notDisposed: notDisposed,
+      notGCed: notGCed,
+      classes: classes,
+    );
+  }
+}
+
+/// Leak tracking settings for tests.
+///
+/// Should be instantiated using static methods of [LeakTrackingForTests].
```

Comment:
Why is that a separate class then instead of just defining these on LeakTrackingForTestsSettings directly?